diff --git a/arrow-buffer/src/alloc/alignment.rs b/arrow-buffer/src/alloc/alignment.rs
index de8a491c8..4dad632fc 100644
--- a/arrow-buffer/src/alloc/alignment.rs
+++ b/arrow-buffer/src/alloc/alignment.rs
@@ -109,3 +109,9 @@ pub const ALIGNMENT: usize = 1 << 6;
 /// Cache and allocation multiple alignment size
 #[cfg(target_arch = "loongarch64")]
 pub const ALIGNMENT: usize = 1 << 6;
+
+// ESP32-S3 uses Xtensa architecture
+// Cache line size is typically 32 bytes for Xtensa
+/// Cache and allocation multiple alignment size
+#[cfg(target_arch = "xtensa")]
+pub const ALIGNMENT: usize = 1 << 5;
